pkg/kubelet/kuberuntime/helpers.go

```diff
@@ -202,8 +202,37 @@ func toKubeRuntimeStatus(status *runtimeapi.RuntimeStatus) *kubecontainer.Runtim
 	return &kubecontainer.RuntimeStatus{Conditions: conditions}
 }
 
+func (m *kubeGenericRuntimeManager) getSeccompProfile(annotations map[string]string, containerName string,
+	podSecContext *v1.PodSecurityContext, containerSecContext *v1.SecurityContext) string {
+	fieldProfile := func(scmp *v1.SeccompProfile) string {
+		if scmp.Type == v1.SeccompProfileUnconfined {
+			return v1.SeccompProfileNameUnconfined
+		}
+		if scmp.Type == v1.SeccompProfileTypeDefault {
+			return v1.SeccompProfileRuntimeDefault
+		}
+		if scmp.Type == v1.SeccompProfileLocalhost && scmp.LocalhostProfile != nil {
+			fname := filepath.Join(m.seccompProfileRoot, *scmp.LocalhostProfile)
+			return v1.SeccompLocalhostProfileNamePrefix + fname
+		}
+
+		return ""
```

Comment:
this path shouldn't be possible, right? what should we do if we get here? what does returning "" do?

Reply:
Returning "" causes dockershim to use the unconfined profile:
https://github.com/kubernetes/kubernetes/blob/a79c711191d5c0a9dca4fbaba26ef35e476f5871/pkg/kubelet/dockershim/helpers_linux.go#L52

I removed this and made the behaviour a bit more explicit.